tests/ra_aid/test_main.py

```
@@ -0,0 +1,123 @@
+
+
+def test_temperature_validation(mock_dependencies):
+    """Test that temperature argument is correctly passed to initialize_llm."""
+    from ra_aid.__main__ import main, initialize_llm
+    import sys
+    from unittest.mock import patch
+
+    _global_memory.clear()
+    
+    with patch('ra_aid.__main__.initialize_llm') as mock_init_llm:
+        with patch.object(sys, 'argv', ['ra-aid', '-m', 'test', '--temperature', '0.7']):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)